verify.go

```diff
@@ -17,53 +17,51 @@ import (
 // IDTokenVerifier provides verification for ID Tokens.
 type IDTokenVerifier struct {
 	keySet *remoteKeySet
-	config *verificationConfig
+	config *Config
+	issuer string
 }
 
-// verificationConfig is the unexported configuration for an IDTokenVerifier.
-//
-// Users interact with this struct using a VerificationOption.
-type verificationConfig struct {
-	issuer string
-	// If provided, this value must be in the ID Token audiences.
-	audience string
-	// If not nil, check the expiry of the id token.
-	checkExpiry func() time.Time
+// Config is the configuration for an IDTokenVerifier.
+type Config struct {
+	// If provided, ClientID must be in the ID Token audiences.
+	// It ensures that an ID Token was issued for the specific client.
+	ClientID string
```

Comment:
So `ClientID` might need to match either `aud` or `azp`: https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation

Do we want to do anything special here? Like say that if there are multiple audiences `ClientID` should match the authorizing party field? Is that too magic?